src/Shouldly/Internals/StringHelpers.cs

```diff
@@ -41,9 +42,17 @@ internal static class StringHelpers
             {
                 var objects = enumerable.Cast<object>();
                 var inspect = "[" + objects.Select(o => o.ToStringAwesomely()).CommaDelimited() + "]";
-                if (inspect == "[]" && value.ToString() != objectType.FullName)
+                if (inspect == "[]")
                 {
-                    inspect += " (" + value + ")";
+                    if (value is IProxy proxy)
```

Comment:
The reason I went with `IProxy` is because, at creation and test time, the type `IEnumerable<T>` is necessary, but `StringHelper` doesn't have or need the type argument, so a non-generic type is ideal. Since this hack doesn't need to be confined to `IEnumerable`, I made the interface a more generic name. I was also wondering thinking of moving it outside of the enumerable-specific logic, but did this for now to get something quick for discussion. If the ability to proxy values has a broader use-case, then I wouldn't mind making it a more generalized approach.

Reply:
Good point, I missed that `IProxy` is currently essential. Its naming convention (it's not `IEnumerableProxy`) makes it look like something I would expect to be able to use for non-enumerable situations, so that is what makes me want this outside the `TryGetEnumerable` check.